src/bidiMapper/domains/network/NetworkRequest.ts

```diff
@@ -237,7 +249,7 @@ export class NetworkRequest {
     return {
       request:
         this.#requestWillBeSentEvent?.requestId ?? NetworkRequest.#unknown,
-      url: this.#requestWillBeSentEvent?.request.url ?? NetworkRequest.#unknown,
+      url: this.url ?? NetworkRequest.#unknown,
```

Comment:
Not sure if this is correct when we have redirects, but I will verify it with my change.

Reply:
Noted